pkg/plugins/manager/dashboard_file_store.go

```diff
@@ -0,0 +1,92 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/util"
+)
+
+var openDashboardFile = func(name string) (fs.File, error) {
+	// Wrapping in filepath.Clean to properly handle
+	// gosec G304 Potential file inclusion via variable rule.
+	return os.Open(filepath.Clean(name))
```

Comment:
@wbrowne btw did you noticed this? I found a way to remove the `nolint:gosec`. Think it makes sense?

Reply:
Yes! I did notice there was no `nolint` alright. But wasn't quite sure why it didn't complain about the risk of the file path argument

Reply:
> But wasn't quite sure why it didn't complain about the risk of the file path argument

Yeah :smile: Question is if we want to do this rather than using `nolint:gosec`. It obviously feels better to not have the `nolint:gosec`, but important to have proper tests and validation. So maybe we can define our own `OpenFile` function that will validate/clean and use this line at last - then maybe we can get rid of `nolint:gosec` in many places? Something we might look in to after this pr. WDYT?

Reply:
That sounds like a great plan 🚀